src/node/src/storage_node_light_impl.rs

```diff
@@ -110,12 +160,124 @@ impl StorageNodeV2Impl {
+                                    warn!("the channel has been closed by client for addr 0x{}", hex::encode(key.as_ref()));
+                                    continue;
+                                }
+                                for idx in 0..sub.topics.len() {
+                                    if sub.topics[idx] != EventTypeV2::Block as i32 {
+                                        continue;
+                                    }
+                                    match sender.try_send(Ok(event.clone())) {
+                                        Ok(_) => {
+                                            info!("send event to addr 0x{}", hex::encode(key.as_ref()));
+                                            break;
+                                        }
+                                        Err(tokio::sync::mpsc::error::TrySendError::Full(_)) => {
+                                            // retry?
+                                            // TODO
+                                            warn!("the channel is full for addr 0x{}", hex::encode(key.as_ref()));
+                                        }
+                                        Err(tokio::sync::mpsc::error::TrySendError::Closed(_)) => {
+                                            // remove the address
+                                            to_be_removed.insert(key.clone());
+                                            warn!("the channel has been closed by client for addr 0x{}", hex::encode(key.as_ref()));
+                                        }
+
+                                    }
+                                }
+                            }
+                        },
+                        else => {
+                            info!("unexpected channel update");
+                            // reconnect in 5 seconds
+                            sleep(TokioDuration::from_millis(1000 * 5)).await;
+                            break;
+                        }
+
+                    }
+                    for k in to_be_removed.iter() {
+                        subscribers.remove(k);
+                    }
+                    to_be_removed.clear();
+                }
+            }
+            info!("exit the keep subscription thread");
+        });
+        Ok(())
+    }
 }
 
 #[tonic::async_trait]
 impl StorageNode for StorageNodeV2Impl {
+    type SubscribeStream = ReceiverStream<std::result::Result<EventMessageV2, Status>>;
+    /// add subscription to the light node
+    async fn subscribe(
+        &self,
+        request: Request<SubscribeRequest>,
+    ) -> std::result::Result<Response<Self::SubscribeStream>, Status> {
+        info!("receive subscribe request");
+        let r = request.into_inner();
+        let sender = self.sender.clone();
+        info!("sender is close: {}", sender.is_closed());
+        let account_id = DB3Verifier::verify(r.payload.as_ref(), r.signature.as_ref())
```

Comment:
https://github.com/dbpunk-labs/db3/blob/main/src/node/src/mutation_utils.rs#L64 
use this scheme to verify the signature

Reply:
will use that once we refactor mutation_util to support general payload verify